src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.cs

```diff
@@ -0,0 +1,455 @@
+
+            var remainingTrivia = _lexer.TryScanRemainingTrivia();
+
+            var eolToken = CurrentCharAsToken().With(leadingTrivia: remainingTrivia.HasValue ? ImmutableArray.Create(remainingTrivia.Value) : ImmutableArray<StackFrameTrivia>.Empty);
+
+            Debug.Assert(_lexer.Position == _lexer.Text.Length);
+            Debug.Assert(eolToken.Kind == StackFrameKind.EndOfLine);
+
+            var root = new StackFrameCompilationUnit(methodDeclaration, fileInformationResult.Value, eolToken);
+
+            return new(_lexer.Text, root);
+        }
+
+        /// <summary>
+        /// Attempts to parse the full method declaration, optionally adding leading whitespace as trivia. Includes
+        /// all of the generic indicators for types, 
+        /// 
+        /// Ex: [|MyClass.MyMethod(string s)|]
+        /// </summary>
+        private ParseResult<StackFrameMethodDeclarationNode> TryParseMethodDeclaration()
+        {
+            var (success, identifierNode) = TryParseNameNode(scanAtTrivia: true);
+            if (!success)
+            {
+                return ParseResult<StackFrameMethodDeclarationNode>.Abort;
+            }
+
+            if (identifierNode is not StackFrameQualifiedNameNode memberAccessExpression)
+            {
+                return ParseResult<StackFrameMethodDeclarationNode>.Abort;
+            }
+
+            var typeArgumentsResult = TryParseTypeArguments();
+            if (!typeArgumentsResult.Success)
+            {
+                return ParseResult<StackFrameMethodDeclarationNode>.Abort;
+            }
+
+            var argumentsResult = ParseMethodParameters();
+            if (!argumentsResult.Success || argumentsResult.Value is null)
+            {
+                return ParseResult<StackFrameMethodDeclarationNode>.Abort;
+            }
+
+            return new(new(memberAccessExpression, typeArgumentsResult.Value, argumentsResult.Value));
+        }
+
+        /// <summary>
+        /// Parses a <see cref="StackFrameNameNode"/> which could either be a <see cref="StackFrameSimpleNameNode"/> or <see cref="StackFrameQualifiedNameNode" />.
+        /// 
+        /// Nodes will be parsed for arity but not generic type arguments.
+        ///
+        /// <code>
+        /// All of the following are valid nodes, where "$$" marks the parsing starting point, and "[|" + "|]" mark the endpoints of the parsed node excluding trivia
+        ///   * [|$$MyNamespace.MyClass.MyMethod|](string s)
+        ///   * MyClass.MyMethod([|$$string|] s)
+        ///   * MyClass.MyMethod([|$$string[]|] s)
+        ///   * [|$$MyClass`1.MyMethod|](string s)
+        ///   * [|$$MyClass.MyMethod|][T](T t)
+        /// </code>
+        /// 
+        /// </summary>
+        private ParseResult<StackFrameNameNode> TryParseNameNode(bool scanAtTrivia)
+        {
+            var currentIdentifer = _lexer.TryScanIdentifier(scanAtTrivia: scanAtTrivia, scanLeadingWhitespace: true, scanTrailingWhitespace: false);
```

Comment:
you always pass scanLeadingWhtispace here.  So please add tests that show we do the right thing with leading whitespace wherever TryParseNameNode can be called.

Reply:
Those exist in a mix between invalid tests, parameter tests, and generic argument tests